x-pack/plugins/threat_intelligence/public/modules/query_bar/components/filter_in/filter_in.tsx

```diff
@@ -102,9 +58,54 @@ export const FilterIn: VFC<FilterInProps> = ({ data, field, type, as: Component,
         iconSize="s"
         size="xs"
         color="primary"
-        onClick={filterIn}
+        onClick={filterFn}
         {...props}
       />
     </EuiToolTip>
   );
 };
+
+/**
+ * Retrieves the indicator's field and value, then creates a new {@link Filter} and adds it to the {@link FilterManager}.
+ *
+ * This component is to be used in an EuiContextMenu.
+ *
+ * @returns filter in item for a context menu
+ */
+export const FilterInContextMenu: VFC<FilterInProps> = ({ data, field, ...props }) => {
+  const { filterFn } = useFilterInOut({ indicator: data, field, filterType: FilterIn });
+  if (!filterFn) {
+    return <></>;
+  }
+
+  return (
+    <EuiContextMenuItem key="filterIn" icon="plusInCircle" size="s" onClick={filterFn} {...props}>
+      Filter In
+    </EuiContextMenuItem>
+  );
+};
+
+/**
+ * Retrieves the indicator's field and value, then creates a new {@link Filter} and adds it to the {@link FilterManager}.
+ *
+ * This component is to be used in an EuiDataGrid.
+ *
+ * @returns filter in button for data grid
+ */
+export const FilterInDataGrid: VFC<FilterInProps> = ({ data, field, as: Component, ...props }) => {
```

Comment:
Is there anything specific to DataGrid? From a first glance, it is a "filter in as icon with a tooltip", does it make sense to name it more generic?

Reply:
This component is meant to be used as a cell actions in the EuiDataGrid. Maybe I could call it `FilterInCellAction` or `FilterInDataGridCellAction`. The `as` or `Component` props is specific to that usage.

Reply:
yeah looking at the EuiDataGrid doc, this is how they describe the `Component` props of the `EuiDataGridColumnCellActionProps` interface:

_React component representing the action displayed in the cell_

Reply:
yeah, I think `FilterInCellAction` is a better name. Now I see where all this extravaganza with `Component` is coming from https://elastic.github.io/eui/#/tabular-content/data-grid-cells-popovers#cell-actions

Reply:
yup I'm going with it, thank you!!

Reply:
changed to `FilterInCellAction`